install and config redesign working branch (#3058)
zowe/docs-site: docs/user-guide/address-security-requirements.md

# Address security requirements

During configuration of server-side components, it is necessary to configure various system security settings. Your
organization may require your security administrator to complete steps to configure Zowe security. As a system
administrator/programmer, first consult with your security administrator before you start the installation process.

**Note:** This article addresses configuring Zowe security during the Zowe z/OS components installation process, and
does not address security configuration to extend Zowe. For more information about security configuration to extend
Zowe, see the following articles:

- [Digital certificates](../getting-started/zowe-security-overview#digital-certificates)
- [User Authentication](../getting-started/zowe-security-overview#user-authentication)
- [Access Authorization](../getting-started/zowe-security-overview#access-authorization)

## Tasks performed by your security administrator

> To configure Zowe security for production environments, your organization's security administrator will be required to

Reviewer: Not just production environments I guess

| Cross memory server (ZIS)                                                                     | [Configuring the cross memory server for SAF](../user-guide/configure-zos-system.md/#configure-the-cross-memory-server-for-saf)<br />and<br />[Configuring cross memory server load module](../user-guide/configure-xmem-server.md/#load-module)<br />and<br />[Configuring cross-memory server SAF configuration](../user-guide/configure-xmem-server.md/#saf-configuration) |

### Assign security permissions of users

As a security administrator, assign users and the ZWEADMIN security group permissions required to perform specific
tasks.

### Overview of user categories and roles

Specific user IDs with sufficient permissions are required to run or access Zowe.
Your organization's security administrator is responsible to assign the following user IDs during Zowe z/OS component
configuration.

The following user IDs run Zowe:

* **[ZWESVUSR](#zwesvusr)**  
  This is the started task ID of the Zowe runtime user who runs most of the Zowe core
  components. <!-- It seems this information about working with USS should be in a table and not in this short description -->

>   To work with USS, this user ID must have a valid OMVS segment. For detailed information about which permissions are

Reviewer: We expect the reader to know what OMVS segment is?

# Address security requirements

> During configuration of service-side components, it is necessary to configure various system security settings. Your organization may require your security administrator to complete steps to configure Zowe security. As a system administrator/programmer, first consult with your security administrator before you start the installation process. 

Reviewer 2: ```suggestion
During configuration of server-side components, it is necessary to configure various system security settings. Your organization may require your security administrator to complete steps to configure Zowe security. As a system administrator/programmer, first consult with your security administrator before you start the installation process. 
```

## Tasks performed by your security administrator

To configure Zowe security for production environments, it is likely that your organization's security administrator will be required to perform various tasks. Some of the tasks apply to general Zowe configuration, while other tasks are required during installation if you plan to use specific Zowe components or features. 

The following required configuration tasks are likely to be performed by your organization's security administrator:

* [Initialize Zowe security configurations](../user-guide/initialize-security-configuration.md)
* [Perform APF autorization of load libraries](../user-guide/apf-authorize-load-library.md)
* [Configure the z/OS system for Zowe](../user-guide/configure-zos-system.md/#configure-user-ids-and-groups-for-the-zowe-started-tasks)
* [Configure address space job naming](../user-guide/configure-zos-system.md/#configure-address-space-job-naming)

If your Zowe server-side installation includes the features listed in the following table, your organization's security administrator may also need to perform the associated tasks:

| Feature of a Zowe server-side component  | Configuration Task  | 
| ---- | ---- | 

> | Use of Top Secret | [Configuring multi-user address space (for TSS only)](../user-guide/configure-zos-system.md/#configure-multi-user-address-space-for-tss-only) |

Reviewer 2: When you are using Top Secret as your security manager

| High Availability | [Configuring ZWESLSTC to run Zowe high availability instances under ZWESVUSR user ID](../user-guide/configure-zos-system.md/#configure-zweslstc-to-run-zowe-high-availability-instances-under-zwesvusr-user-id) |
| z/OSMF authentication or onboarding of z/OSMF service | [Granting users permission to access z/OSMF](../user-guide/grant-user-permission-zosmf.md) |
| ZSS component enabled (required for API ML certificate and identity mapping) |[Configuring an ICSF cryptographic services environment](../user-guide/configure-zos-system.md/#configure-an-icsf-cryptographic-services-environment) <br />and<br /> [Configuring security environment switching](../user-guide/configure-zos-system.md/#configure-security-environment-switching)|
| API Mediation Layer certificate mapping | [Configuring main Zowe server to use client certificate identity mapping](../user-guide/configure-zos-system.md/#configure-main-zowe-server-to-use-client-certificate-identity-mapping) |
| API Mediation Layer identity mapping | [Configuring main Zowe server to use distributed identity mapping](../user-guide/configure-zos-system.md/#configure-main-zowe-server-to-use-distributed-identity-mapping) |

> | API Mediation Layer Identity tokens (IDT) | [Configuring signed SAF Identity tokens (IDT)](../user-guide/configure-zos-system.md/#configure-signed-saf-identity-tokens-idt) |

Reviewer 2: ```suggestion
| API Mediation Layer Identity Tokens (IDT) | [Configuring signed SAF Identity tokens (IDT)](../user-guide/configure-zos-system.md/#configure-signed-saf-identity-tokens-idt) |
```

| API Mediation Layer certificate mapping | [Configuring main Zowe server to use client certificate identity mapping](../user-guide/configure-zos-system.md/#configure-main-zowe-server-to-use-client-certificate-identity-mapping) |
| API Mediation Layer identity mapping | [Configuring main Zowe server to use distributed identity mapping](../user-guide/configure-zos-system.md/#configure-main-zowe-server-to-use-distributed-identity-mapping) |
| API Mediation Layer Identity Tokens (IDT) | [Configuring signed SAF Identity tokens (IDT)](../user-guide/configure-zos-system.md/#configure-signed-saf-identity-tokens-idt) |
| Cross memory server (ZIS) | [Configuring the cross memory server for SAF](../user-guide/configure-zos-system.md/#configure-the-cross-memory-server-for-saf)<br />and<br />[Configuring cross memory server load module](../user-guide/configure-xmem-server.md/#load-module)<br />and<br />[Configuring cross-memory server SAF configuration](../user-guide/configure-xmem-server.md/#saf-configuration) |

> ## Assign security permissions of users

Reviewer 2: These are also tasks for sec admin. The title should not be on the same level.

Reviewer 3: resolved

As a security administrator, assign users and the ZWEADMIN security group permissions required to perform specific tasks. 

### Overview of user categories and roles

Specific user IDs with sufficient permissions are required to run or access Zowe. 
Your organization's security administrator is responsible to assign the following user IDs during Zowe z/OS component configuration.

> The following user IDs can run Zowe: 

Reviewer 2: ```suggestion
The following user IDs run Zowe: 
```

The following reference table describes which permissions are required for the user ID `ZWESVUSR` to run Zowe core services and specific individual components.

If you already successfully ran the [`ZWESECUR`](/docs/user-guide/initialize-security-configuration.md#configuring-with-zwesecur-jcl) JCL either separately or by running the [`zwe init security`](/docs/user-guide/initialize-security-configuration.md#configuring-with-zwe-init-security-command) command, you do not need to perform the steps described in this section. The TSO commands to create the user IDs and groups are executed during the JCL sections of `ZWESECUR`. For more information about the `zwe init security` command, see [zwe init security](../appendix/zwe_server_command_reference/zwe/init/zwe-init-security.md).

| Feature of a Zowe server-side component                    | Resource class    | Resource name                          | Type of access required | Reason | Actions                                                                                                                                                                                                                        |

> |------------------------------------------------------------|----------|-----------------------------|--------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|-----|

Reviewer 2: I think this is good idea

> | Cross memory server (ZIS)                                  | CSFSERV                                 | `Multiple`                  | READ   | Generate symmetric keys using ICSF that is used by [Zowe Desktop cookies](./configure-zos-system.md#configure-an-icsf-cryptographic-services-environment). | The list of IDs to enable include `CSF1TRD` , `CSF1TRC` , `CSF1SKE` , `CSF1SKD`. The full list of IDs is described in the z/OS Cryptographic Services user guide for your z/OS release level: [2.2](https://www.ibm.com/docs/en/zos/2.2.0?topic=ssl-racf-csfserv-resource-requirements), [2.3](https://www.ibm.com/docs/en/zos/2.3.0?topic=ssl-racf-csfserv-resource-requirements), [2.4](https://www.ibm.com/docs/en/zos/2.4.0?topic=ssl-racf-csfserv-resource-requirements) and [2.5](https://www.ibm.com/docs/en/zos/2.5.0?topic=ssl-racf-csfserv-resource-requirements). |

Reviewer 2: I would not lable it core since many customers are using Zowe without ZSS/ZIS. 

Core should be permissions that are mandatory and zowe will not run without them.

> | Cross memory server (ZIS)                                  | CSFSERV                                 | `Multiple`                  | READ   | Generate symmetric keys using ICSF that is used by [Zowe Desktop cookies](./configure-zos-system.md#configure-an-icsf-cryptographic-services-environment). | The list of IDs to enable include `CSF1TRD` , `CSF1TRC` , `CSF1SKE` , `CSF1SKD`. The full list of IDs is described in the z/OS Cryptographic Services user guide for your z/OS release level: [2.2](https://www.ibm.com/docs/en/zos/2.2.0?topic=ssl-racf-csfserv-resource-requirements), [2.3](https://www.ibm.com/docs/en/zos/2.3.0?topic=ssl-racf-csfserv-resource-requirements), [2.4](https://www.ibm.com/docs/en/zos/2.4.0?topic=ssl-racf-csfserv-resource-requirements) and [2.5](https://www.ibm.com/docs/en/zos/2.5.0?topic=ssl-racf-csfserv-resource-requirements). |

Reviewer 4: Are you sure that Zowe will not use CSF for things like generating and reading keyrings?
I'm not sure, I think it might, so that's why I said core.

> | Cross memory server (ZIS)                                  | CSFSERV                                 | `Multiple`                  | READ   | Generate symmetric keys using ICSF that is used by [Zowe Desktop cookies](./configure-zos-system.md#configure-an-icsf-cryptographic-services-environment). | The list of IDs to enable include `CSF1TRD` , `CSF1TRC` , `CSF1SKE` , `CSF1SKD`. The full list of IDs is described in the z/OS Cryptographic Services user guide for your z/OS release level: [2.2](https://www.ibm.com/docs/en/zos/2.2.0?topic=ssl-racf-csfserv-resource-requirements), [2.3](https://www.ibm.com/docs/en/zos/2.3.0?topic=ssl-racf-csfserv-resource-requirements), [2.4](https://www.ibm.com/docs/en/zos/2.4.0?topic=ssl-racf-csfserv-resource-requirements) and [2.5](https://www.ibm.com/docs/en/zos/2.5.0?topic=ssl-racf-csfserv-resource-requirements). |

Reviewer 2: Keyrings are generated under a security admin ID. This table describes permission needed from a Zowe server user.

| Cross memory server (ZIS)                                  | FACILITY                                | `ZWES.IS`                   | READ   | To allow Zowe ZWESLSTC processes to access the Zowe ZIS cross memory server.  | Run the command to permit Zowe main server to use ZIS cross memory server.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  |

> | ZSS (required for API ML certificate and identity mapping) | FACILITY                                | `BPX.SERVER` + `BPX.DAEMON` | UPDATE | Allows Zowe to run code on behalf of the API requester's TSO user ID. For more information, see [Security Environment Switching](./configure-zos-system.md#configure-security-environment-switching). | Run the commands to permit the Zowe main server to create a user's security environment |                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     |

Reviewer 4: Simplify to "ZSS". There are various reasons why you'd be required to use any component. This column should not be bloated with all those details.

Context:
This row is required by ZSS, a default component. ZSS can exist with or without API ML. Saying that ZSS is required by API ML in certain circumstances is a detail seems like something that instead should be present in the API ML certificate & identity mapping documentation.

Reviewer 4: Keep in mind there's more dependencies, so it could be quite a complex table if you want to fit it all here.

Reviewer 2: I agree. This is one of the biggest configuration challenges that I am currently seeing. Admins are not aware of dependencies between components. I tried to address this feedback at least partially here.

| Core                                                       | FACILITY                                                    | `BPX.JOBNAME`               | READ   | To allow z/OS address spaces for unix processes to be renamed for [ease of identification](./configure-zos-system.md#configure-address-space-job-naming). | Run the commands to permit the Zowe main server to set the job name.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      |

> | Core?                                                      | FACILITY                                                   | `IRR.RADMIN.LISTUSER`       | READ   | To allow Zowe to obtain information about OMVS segment of the user profile using `LISTUSER` TSO command.   | Run the commands to permit Zowe ... |                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     |

Reviewer 2: I don't know what this is good for. If no one else clarifies then just remove '?'.

  - [Digital certificates](../getting-started/zowe-security-overview#digital-certificates)
  - [User Authentication](../getting-started/zowe-security-overview#user-authentication)
  - [Access Authorization](../getting-started/zowe-security-overview#access-authorization)

## Tasks performed by your security administrator

In order to configure security for production environments, various tasks need to be performed by your organization's security administrator. Some of the tasks apply to general Zowe configuration, while other tasks are required only if specific features are installed. 

The following required configuration tasks are likely to be performed by your organization's security administrator:

* [Initialize Zowe security configurations](../user-guide/initialize-security-configuration.md)
* [Perform APF autorization of load libraries](../user-guide/apf-authorize-load-library.md)
* [Configure the z/OS system for Zowe](../user-guide/configure-zos-system/#configure-user-ids-and-groups-for-the-zowe-started-tasks)
* [Configure address space job naming](../user-guide/configure-zos-system/#configure-address-space-job-naming)

> * [Use certificates](../user-guide/use-certificates.md)

Reviewer 2: I would rather link the first certificate page ... probably an overview.

If your installation of Zowe includes the features listed in the following table, your organization's security administrator may also need to perform the associated tasks:

| Feature of a Zowe server-side component  | Configuration Task  | 
| ---- | ---- | 

> | z/OSMF authentication and service | [Granting users permission to access z/OSMF](../user-guide/grant-user-permission-zosmf.md) |
> | ZSS, API ML certificate mapping, and identity mapping |[Configuring an ICSF cryptographic services environment](../user-guide/configure-zos-system.md/#configure-an-icsf-cryptographic-services-environment) |
> | TSS | [Configuring multi-user address speace (for TSS only)](../user-guide/configure-zos-system.md/#configure-multi-user-address-space-for-tss-only) |
> | High Availability | [Configuring ZWESLSTC to run Zowe high availability instances under ZWESVUSR user ID](../user-guide/configure-zos-system.md/#configure-zweslstc-to-run-zowe-high-availability-instances-under-zwesvusr-user-id) |
> | Cross memory server for SAF | [Configuring the cross memory server for SAF](../user-guide/configure-zos-system.md/#configure-the-cross-memory-server-for-saf) |
> | ZSS, API ML certificate mapping and identity mapping | [Configuring security environment switching](../user-guide/configure-zos-system.md/#configure-security-environment-switching) |
> | API Mediation Layer certificate mapping | [Configuring main Zowe server to use client certificate identity mapping](../user-guide/configure-zos-system.md/#configure-main-zowe-server-to-use-client-certificate-identity-mapping) |
> | API Mediation Layer identity mapping | [Configuring main Zowe server to use distributed identity mapping](../user-guide/configure-zos-system.md/#configure-main-zowe-server-to-use-distributed-identity-mapping) |
> | API Mediation Layer Identity tokens (IDT) | [Configuring signed SAF Identity tokens (IDT)](../user-guide/configure-zos-system.md/#configure-signed-saf-identity-tokens-idt) |
> | Cross memory server load module | [Configuring cross memory server load module](../user-guide/configure-xmem-server.md/#load-module) |
> | Cross memory server SAF | [Configuring cross-memory server SAF configuration](../user-guide/configure-xmem-server.md/#saf-configuration) |

Reviewer 2: ```suggestion
| z/OSMF authentication or onboarding of z/OSMF service | [Granting users permission to access z/OSMF](../user-guide/grant-user-permission-zosmf.md) |
| ZSS component enabled (required for API ML certificate and identity mapping) |[Configuring an ICSF cryptographic services environment](../user-guide/configure-zos-system.md/#configure-an-icsf-cryptographic-services-environment) |
| When Top Secret is used | [Configuring multi-user address space (for TSS only)](../user-guide/configure-zos-system.md/#configure-multi-user-address-space-for-tss-only) |
| High Availability | [Configuring ZWESLSTC to run Zowe high availability instances under ZWESVUSR user ID](../user-guide/configure-zos-system.md/#configure-zweslstc-to-run-zowe-high-availability-instances-under-zwesvusr-user-id) |
| Cross memory server (XMEM) | [Configuring the cross memory server for SAF](../user-guide/configure-zos-system.md/#configure-the-cross-memory-server-for-saf) |
| ZSS component enabled (required for API ML certificate and identity mapping) | [Configuring security environment switching](../user-guide/configure-zos-system.md/#configure-security-environment-switching) |
| API Mediation Layer certificate mapping | [Configuring main Zowe server to use client certificate identity mapping](../user-guide/configure-zos-system.md/#configure-main-zowe-server-to-use-client-certificate-identity-mapping) |
| API Mediation Layer identity mapping | [Configuring main Zowe server to use distributed identity mapping](../user-guide/configure-zos-system.md/#configure-main-zowe-server-to-use-distributed-identity-mapping) |
| API Mediation Layer Identity tokens (IDT) | [Configuring signed SAF Identity tokens (IDT)](../user-guide/configure-zos-system.md/#configure-signed-saf-identity-tokens-idt) |
| Cross memory server (XMEM) | [Configuring cross memory server load module](../user-guide/configure-xmem-server.md/#load-module) |
| Cross memory server (XMEM) | [Configuring cross-memory server SAF configuration](../user-guide/configure-xmem-server.md/#saf-configuration) |
```

Reviewer 2: Let's meet and discuss the order.

### User ID requirements and security permissions

Specific user IDs with sufficient permissions are required to run or access Zowe. Review the following table to ... <!-- Provide details about the task and the specific role to perform the task -->

### ZWESVUSR

This is a started task ID for `ZWESLSTC`.  

The task starts a USS environment using `BPXBATSL` that executes the core Zowe Desktop (ZLUX) node.js server, the Java API Mediation Layer, and the Z Secure Services C component.  To work with USS, the user ID `ZWESVUSR` must have a valid OMVS segment.  

> | Class    | ID                          | Access | Reason                                                                                                                                                                                                                        |

Reviewer 2: ```suggestion
| Resource class    | Resource name                          | Type of access required | Reason                                                                                                                                                                                                                        |
```

## Assign security permissions of users

As a security administrator, assign users and the ZWEADMIN security group permissions required to perform specific tasks. 

### Overview of user categories and roles

Specific user IDs with sufficient permissions are required to run or access Zowe. 
Your organization's security administrator is responsible to assign the following user IDs during Zowe z/OS component configuration.

The following user IDs can run Zowe: 

* **[ZWESVUSR](#zwesvusr)**  
This user runs most of the Zowe core components.
* **[ZWESIUSR](#zwesiusr)**  
This user runs the cross memory server (ZIS)

The security administrator also assigns permissions to the security group [ZWEADMIN](#zweadmin), as well as permissions to individual Zowe users.

### ZWESVUSR

This is a started task ID for `ZWESLSTC`.  

> The task starts a USS environment using `BPXBATSL` that executes the core Zowe Desktop (ZLUX) node.js server, the Java API Mediation Layer, and the Z Secure Services C component.  To work with USS, the user ID `ZWESVUSR` must have a valid OMVS segment.  

Reviewer 2: This does not seem to belong here. It is a technical detail with no meaning for a security admin. But I have no idea where it belongs

### ZWESIUSR

This is a started task ID used to run the PROCLIB `ZWESISTC` that launches the [cross memory server (ZIS)](./configure-xmem-server.md). This started task ID must have a valid OMVS segment.

### ZWEADMIN

ZWEADMIN is a group that `ZWESVUSR` and `ZWESIUSR` should belong to. This group must have a valid OMVS segment.  

### zowe_user

If z/OSMF is used for authentication and serving REST APIs for Zowe CLI and Zowe Explorer users, the TSO user ID for end users must belong to one or both of the groups `IZUUSER` or `IZUADMIN`.

> ## Security Permissions Reference Table 

Reviewer 2: I am not certain but I think this table describes permissions only for ZWESVUSR.

As a ... (Identify the role/roles who are addressing these requirements. We should include an **Important:** note to inform readers in which use case they are likely to need permissions set by their security administrator.)

There are two parts of configuring security:
* Configuring Zowe security

> * Extending Zowe

Reviewer: Extending sounds as if you need to read it only when you are extending Zowe. 

What about something like: Additional Security Concepts?

| API Mediation Layer certificate mapping | [Configuring main Zowe server to use client certificate identity mapping](../user-guide/configure-zos-system.md/#configure-main-zowe-server-to-use-client-certificate-identity-mapping) |
| API Mediation Layer identity mapping | [Configuring main Zowe server to use distributed identity mapping](../user-guide/configure-zos-system.md/#configure-main-zowe-server-to-use-distributed-identity-mapping) |
| API Mediation Layer Identity Tokens (IDT) | [Configuring signed SAF Identity tokens (IDT)](../user-guide/configure-zos-system.md/#configure-signed-saf-identity-tokens-idt) |
| Cross memory server (ZIS) | [Configuring the cross memory server for SAF](../user-guide/configure-zos-system.md/#configure-the-cross-memory-server-for-saf)<br />and<br />[Configuring cross memory server load module](../user-guide/configure-xmem-server.md/#load-module)<br />and<br />[Configuring cross-memory server SAF configuration](../user-guide/configure-xmem-server.md/#saf-configuration) |

> ## Assign security permissions as required

Reviewer 2: ```suggestion
## Assign security permissions to users as required
```

As a security administrator, assign users and the ZWEADMIN security group permissions required to perform specific tasks. 

**Note:** Some permissions are required to run Zowe core, while other permissions are required to run specific components. 
### Overview of user security permissions

Specific user IDs with sufficient permissions are required to run or access Zowe. 

> As a security administrator, you are responsible to assign the following user IDs during Zowe z/OS component configuration.

Reviewer 2: ```suggestion
As a security administrator, you are responsible to assign security permission to the following user IDs during Zowe z/OS component configuration.
```

The following user IDs can run Zowe: 

* **[ZWESVUSR](#zwesvusr)**  

> This is the stated task ID of the Zowe runtime user who runs most of the Zowe core components. <!-- It seems this information about working with USS should be in a table and not in this short description -->To work with USS, this user ID must have a valid OMVS segment. For detailed information about which permissions are required to run Zowe core services as well as specific individual components, see the [Security Permissions Reference Table](#security-permissions-reference-table) in this article.

Reviewer: started instead of stated

Reviewer 3: resolved
